presto-main/src/main/java/com/facebook/presto/operator/ArrayUnnester.java

```diff
@@ -35,17 +33,17 @@
     private int position;
     private final int positionCount;
 
-    public ArrayUnnester(ArrayType arrayType, @Nullable Slice slice)
+    public ArrayUnnester(ArrayType arrayType, @Nullable Block arrayBlock)
     {
         this.channelCount = 1;
         this.elementType = checkNotNull(arrayType, "arrayType is null").getElementType();
 
-        if (slice == null) {
-            arrayBlock = null;
+        if (arrayBlock == null) {
+            this.arrayBlock = null;
```

Comment:
you can move the `this.arrayBlock = ...` out of the if and just have `this.arrayBlock = arrayBlock`